lib/rules/indent.js

```diff
@@ -375,16 +375,39 @@ module.exports = {
         }
 
         /**
+         * Returns a boolean if the parent type is in the nodeList
+         * @param {string} parentType type that is being checked in nodeList
+         * @param {array} nodeList list of node types passed in
+         * @returns {boolean} if parentType matches item in nodeList, otherwise false
+         */
+        function checkIfParentTypeIsInStopNodeList(parentType, nodeList) {
```

Comment:
@not-an-aardvark implemented what you recommended, with a small tweak.  With the current logic in the whole loop we would need to change it to `nodeList.indexOf(parentType) === -1` because in the loop we are looping until `parent === type, stopAtList.indexOf(parent.type) === 1, and parent.type === 'Program'`.  The semantics get a little tricky with the booleans and my implementation in the loop.  Let me know if I need to explain further or I missed something.

Thank you